packages/wallet/backend/migrations/20240801155211_update_asset_scale.js

```diff
@@ -0,0 +1,112 @@
+const Knex = require('knex')
+const knexConfig = require('../knexfile')
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex) {
+  if (process.env.NODE_ENV !== 'test') {
+    const rafikiKnex = Knex(knexConfig.rafiki_backend)
+
+    try {
+      const assets = await rafikiKnex('assets').distinct('code')
+
+      for (const asset of assets) {
+        const { code } = asset
+
+        const existingAssetWithScale9 = await rafikiKnex('assets')
+          .where({ code, scale: 9 })
+          .first()
+
+        if (existingAssetWithScale9) {
+          await knex('accounts')
+            .where({ assetCode: code, assetScale: 2 })
+            .update({
+              assetId: existingAssetWithScale9.id,
+              assetScale: 9
+            })
+        } else {
+          await rafikiKnex('assets')
+            .where({ code, scale: 2 })
+            .update({ scale: 9 })
+        }
+      }
+    } finally {
+      await rafikiKnex.destroy()
+    }
+  }
+
+  const walletAddresses = await knex('walletAddresses').select(
+    'id',
+    'accountId'
+  )
+  for (const walletAddress of walletAddresses) {
+    const account = await knex('accounts')
+      .where('id', walletAddress.accountId)
+      .first()
+    if (account) {
+      await knex('walletAddresses')
+        .where('id', walletAddress.id)
+        .update({ assetCode: account.assetCode })
```

Comment:
why would asset code change?

Reply:
This happens because `assetCode` was `null` for every WA for some reason and that was causing `keepBalancesSynced` to fail when checking that asset code exists for every WA. Until now, we only populated `assetCode` column on WAs table for WM payment pointers.